src/components/Flyout/useContainScroll.ts

```diff
@@ -0,0 +1,46 @@
+import { MutableRefObject, useEffect } from "react";
+
+export const useContainScroll = (overlayRef: MutableRefObject<HTMLDivElement | null>, { isDisabled = false }) => {
+    useEffect(() => {
+        const overlay = overlayRef.current;
+
+        const scrollHandler = (event: WheelEvent) => {
+            const prevent = () => {
+                event.stopPropagation();
+                event.preventDefault();
+            };
+
+            if (overlay !== null) {
+                const deltaY = event.deltaY;
+                const up = deltaY < 0;
+                const scrollTop = overlay.scrollTop;
+                const scrollHeight = overlay.scrollHeight;
+                const height = overlay.clientHeight;
+
+                if (!up && deltaY > scrollHeight - height - scrollTop) {
+                    // Scrolling down, but this will take us past the bottom.
+                    overlay.scrollTop = scrollHeight;
+                    prevent();
+                    return false;
+                } else if (up && -deltaY > scrollTop) {
+                    // Scrolling up, but this will take us past the top.
+                    overlay.scrollTop = 0;
+                    prevent();
+                    return false;
+                }
+            }
+        };
+
+        const eventProps = { passive: false, capture: true };
+
+        if (!isDisabled && overlay !== null) {
```

Comment:
You could move the eventProps to the top of the method and move this inside the `if` block from above so you don't have to do the overlay check again

Reply:
The other `if` block is inside of the event handler function, so I can't really do that here. An alternative would be to instantiate the `scrollEvent` variable outside of an if, and then assign it the function from within the `if` statement, so that the return function has a reference to it and i don't need to run the overlay check again. 
```
let scrollEvent;
if(overlay !== null){
  scrollEvent = (event) => ...
  window.addEventListener('wheel', scrollEvent)
}
```
I'm not sure if this is any better than what is already there, but wdyt?

Reply:
You're right, I missed that fact! I'm happy with the code, nice work 💪🏽